### PR TITLE
LPS-134948 Fix sorting by latest created in Questions

### PR DESCRIPTION
**Pull request by javierdearcos**

### CODE REVIEW

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/questions/Questions.es.js [hunk 1]
```diff
@@ -310,7 +310,19 @@ export default withRouter(
...
+					currentTag,
+					page,
+					pageSize,
+					search,
+					section,
+					siteKey,
+					'dateCreated:desc'
+				);
+			}
+			else if (filter === 'latest-edited') {
```
Comment (nhpatt) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/questions/Questions.es.js:325]: Can you just add 'dateCreated:desc' as the default value in the last else call?

Comment (nhpatt) [reply modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/questions/Questions.es.js:325]: I don't have a problem of making it explicit but I would remove the else block if there is already an explicit default.

Comment (javierdearcos) [reply modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/questions/Questions.es.js:325]: Sure! I doubt about it, but I was concern that we added more cases in the future that may interfere

**Comment by javierdearcos** [comment]
Updated making the latest-created filter the default filter case

**Review by nhpatt — APPROVED** [review]

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
Uhm... questions failures... let's see if they are a fluke or real.

**Comment by nhpatt** [comment]
ci:test:relevant

**Comment by javierdearcos** [comment]
Checking

**Comment by javierdearcos** [comment]
After fixing this bug, I'm stuck in some bad behavior that I found. I will explain it in the related lines of code

**Comment by javierdearcos** [comment]
Sometimes when we change the filter selector, the GraphQL query resolves with the loading variable as true and causes the app to get stuck with the loading spinner.

I've tried setting the setLoading value to false, ignoring the loading variable from the graphql response, but then sometimes the data items doesn't render...

https://github.com/liferay-frontend/liferay-portal/blob/b63cf4636a1fc0c1210e911669a0ded9ae8150df/modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/questions/Questions.es.js#L355

**Comment by nhpatt** [comment]
There isn't a loading property returned from calling the client manually, see: https://github.com/nearform/graphql-hooks

Removing and setting it to false manually. How can I reproduce the problem rendering the response?

**Comment by javierdearcos** [comment]
Ok, mistery solved. I think that on Friday, when I did the refactor to remove the loading parameter of the then callback, I left the data as the whole object `(data)` instead of leaving it like `({data})` to get the data property inside the returned result.

I'll make a final test and ask you for a final review

**Comment by javierdearcos** [comment]
ci:test:sf

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/questions/Questions.es.js [hunk 2]
```diff
@@ -345,20 +345,21 @@ export default withRouter(
...
-				setLoading(loading);
-			}).catch((error) => {
-				if (process.env.NODE_ENV === 'development') {
-					console.error(error);
-				}
-				setLoading(false);
-				setError({message: 'Loading Questions', title: 'Error'});
-			});
+			})
+				.catch((error) => {
```
Comment (javierdearcos) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/questions/Questions.es.js:356]: I don't like the formatting at all but seems that is like the Source Formatting does with the finally

**Review by nhpatt — APPROVED** [review]

**Comment by nhpatt** [comment]
ci:forward

**Comment by javierdearcos** [comment]
ci:forward

**Comment by javierdearcos** [comment]
ci:forward

**Pull request closed without merging**